roles/manage_agents/tasks/main.yml

```diff
@@ -0,0 +1,20 @@
+---
+- name: List agents currently labeled with the cluster order label
+  kubernetes.core.k8s_info:
+    kind: Agent
+    api_version: agent-install.openshift.io/v1beta1
+    namespace: "{{ default_agent_namespace }}"
+    label_selectors:
+      - "{{ cluster_order_label }}={{ manage_agents_cluster_order_name }}"
+      - "{{ esi_agent_resource_class_label }}={{ manage_agents_resource_class }}"
+  register: manage_agents_allocated
+
+- name: Remove agents from the cluster
+  ansible.builtin.include_tasks:
+    file: remove_agents.yml
+  when: (manage_agents_desired_count | int) < (manage_agents_allocated.resources | length)
```

Comment:
Just curious: These casts to `int` are harmless, but are they necessary? Especially since we're running with `jinja2_native=True`.

Reply:
Not sure we need to cast everywhere, but I remember I had some issues in the past, even with `jinja2_native`, where we sometimes still need to hint jinja that we want a `int` out of a variable.